clippy_lints/src/consts.rs

```diff
@@ -441,12 +442,12 @@ pub fn miri_to_const<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, result: &ty::Const<'
             // FIXME: implement other conversion
             _ => None,
         },
-        ConstValue::ScalarPair(Scalar::Ptr(ptr), Scalar::Bits { bits: n, .. }) => match result.ty.sty {
+        ConstValue::Slice(Scalar::Ptr(ptr), n) => match result.ty.sty {
             ty::Ref(_, tam, _) => match tam.sty {
                 ty::Str => {
                     let alloc = tcx.alloc_map.lock().unwrap_memory(ptr.alloc_id);
                     let offset = ptr.offset.bytes().try_into().expect("too-large pointer offset");
-                    let n = n as usize;
+                    let n = usize::try_from(n).unwrap();
```

Comment:
Calling `try_from()` just to instantly unwrap it looks odd.

Reply:
Yea it is, but it's similar to `foo.checked_add(bar).unwrap()` which prevents silent overflow irrelevant of the compiler flags.